atc/worker/client.go

```diff
@@ -621,7 +621,9 @@ func (client *client) chooseTaskWorker(
 			workerSpec,
 			strategy,
 		); err != nil {
-			return nil, err
+			if _, ok := err.(NoWorkerFitContainerPlacementStrategyError); !ok {
+				return nil, err
+			}
 		}
 
 		if !strategy.ModifiesActiveTasks() {
```

Comment:
When `NoWorkerFitContainerPlacementStrategyError` occurs, it won't break, but `chosenWorker` is nil, so this check should change to `if chosenWorker != nil && !strategy.ModifiesActiveTasks()`.